lib/reply.js

```diff
@@ -617,11 +635,38 @@ function onSendEnd (reply, payload) {
     return
   }
 
+  // node:stream
   if (typeof payload.pipe === 'function') {
     sendStream(payload, res, reply)
     return
   }
 
+  // node:stream/web
+  if (typeof payload.getReader === 'function') {
+    sendWebStream(payload, res, reply)
+    return
+  }
+
+  // Response
+  if (Object.prototype.toString.call(payload) === '[object Response]') {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Response/headers
+    if (typeof payload.headers === 'object' && typeof payload.headers.forEach === 'function') {
+      payload.headers.forEach((headerValue, headerName) => {
+        reply.header(headerName, headerValue)
+      })
```

Comment:
Have you considered using `headers.entries` and iterating over them using `for...of`?
We might reduce the need for a closure here, but not sure how does this might impact performance.

Reply:
Nice, thanks for that 👍
Never benchmarked them before, I would expect better results with the iterator, they are quite interesting

Reply:
`Headers` is not a native implementation same as broswer and it is not extending `Map`.

`.entries` is re-creating the object two times.
https://github.com/nodejs/undici/blob/main/lib/fetch/headers.js#L245-L255

`.forEach` is more simple and do not create the object in middle.

Reply:
`for...of` is slightly faster at best or they are the same

`.entries` creates a bunch of arrays though

Reply:
Double check the implementation inside `undici`, seems like I can directly use `for ... of` as the iterator mapped to `.entries`.

Reply:
It will build a iterator per-spec, have you benchmarked that already?

Reply:
Don't know why, but `for ... of` is faster than `for (const of .enteries())` and `.forEach`.

Reply:
Good learning, I imagine that is as per what @gurgunday suggests. I'd take a look at `undici` later. Thanks for the info